### PR TITLE
Remove hard coded in generator-base-private and add tests on it

### LINKED ISSUE #12870: Remove hard coded path in updateLanguagesInWebpackAngular
##### **Overview of the issue**

To allow blueprint customization we need to use `${this.CLIENT_MAIN_SRC_DIR}` here 
https://github.com/jhipster/generator-jhipster/blob/main/generators/generator-base-private.js#L343

##### **Related issues**

https://github.com/jhipster/generator-jhipster/pull/12410

##### **Suggest a Fix**

I can provide a PR to fix this if you are ok 

##### **Browsers and Operating System**

-   [x] Checking this box is mandatory (this is just to show you read everything)

### PR DESCRIPTION
**Pull request by nicolas63**
---

Please make sure the below checklist is followed for Pull Requests.

-   [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [ ] Tests are added where necessary
-   [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) was updated if necessary
-   [ ] Documentation is added/updated where necessary
-   [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

fix #12870

### CODE REVIEW

File: test/enforcements.spec.js [hunk 1]
```diff
@@ -81,4 +82,29 @@ describe('Enforce some developments patterns', () => {
...
+            describe(`file ${path.basename(file)}`, () => {
+                let content;
+                before(() => {
+                    content = fse.readFileSync(file, 'utf-8');
+                });
+
+                [
+                    ['src/main/webapp', '<%= CLIENT_MAIN_SRC_DIR %>'],
+                    ['src/test/javascript', '<%= CLIENT_TEST_SRC_DIR %>'],
+                ].forEach(([notSpected, replacement]) => {
```
Comment (mshima) [inline test/enforcements.spec.js:101]:
```suggestion
                
                ['src/main/webapp', 'src/test/javascript'].forEach(notSpected => {
```

**Review by mshima — CHANGES_REQUESTED** [review]

File: test/enforcements.spec.js (hunk 1 above)
Comment (nicolas63) [reply test/enforcements.spec.js:101]: done

**Review by mshima — APPROVED** [review]
LGTM

**Comment by mshima** [comment]
Thanks @nicolas63

**Pull request merged**